packages/core/src/storage/didcomm/DidCommMessageRecord.ts

```diff
@@ -58,13 +58,13 @@ export class DidCommMessageRecord extends BaseRecord<DefaultDidCommMessageTags>
   }
 
   public getTags() {
-    const messageId = this.message['@id'] as string
-    const messageType = this.message['@type'] as string
+    const messageId = (this.message['@id'] ?? this.message['id']) as string
+    const messageType = (this.message['@type'] ?? this.message['type']) as string
 
     const { protocolName, protocolMajorVersion, protocolMinorVersion, messageName } = parseMessageType(messageType)
 
     const thread = this.message['~thread']
-    let threadId = messageId
+    let threadId = (this.message['thid'] ?? messageId) as string
```

Comment:
Should we split this in if (didcommv1) {} else if (didcommv2) {} ?

to not make it so intertwined. As it could lead to some issues if for some magic reason a didcomm v2 message has a `~thread` header 😆